environment/k8s_helm_resource.go

```diff
@@ -0,0 +1,243 @@
+	if _, err := action.NewUninstall(k.actionConfig).Run(k.releaseName); err != nil {
+		return err
+	}
+	return nil
+}
+
+// NewReorgGethHelmResource creates new helm manifest for multi-node Geth network
+func NewReorgGethHelmResource() *K8sHelmManifest {
+	return &K8sHelmManifest{
+		id:          "evm",
+		chartPath:   filepath.Join(tools.ProjectRoot, "environment/charts/geth-reorg"),
+		releaseName: "reorg-1",
+		values:      make(map[string]interface{}),
+		SetValuesHelmFunc: func(k *K8sHelmManifest) error {
+			details, err := k.ServiceDetails()
+			if err != nil {
+				return err
+			}
+			for _, d := range details {
+				if d.RemoteURL.Port() == strconv.Itoa(EVMRPCPort) {
+					k.values["clusterURL"] = d.RemoteURL.String()
+					k.values["localURL"] = d.LocalURL.String()
+				}
+			}
+			k.values["rpcPort"] = EVMRPCPort
+			return nil
+		},
+	}
+}
+
+func (k *K8sHelmManifest) ID() string {
+	return k.id
+}
+
+func (k *K8sHelmManifest) SetValue(key string, val interface{}) {
+	k.values[key] = val
+}
+
+func (k *K8sHelmManifest) GetConfig() *config.Config {
+	return k.config
+}
+
+func (k *K8sHelmManifest) Values() map[string]interface{} {
+	return k.values
+}
+
+func (k *K8sHelmManifest) SetEnvironment(
+	k8sClient *kubernetes.Clientset,
+	k8sConfig *rest.Config,
+	config *config.Config,
+	network *config.NetworkConfig,
+	namespace *coreV1.Namespace,
+) error {
+	k.k8sClient = k8sClient
+	k.k8sConfig = k8sConfig
+	k.config = config
+	k.network = network
+	k.namespace = namespace
+	return nil
+}
+
+func (k *K8sHelmManifest) forwardAllPodsPorts() error {
+	k8sPods := k.k8sClient.CoreV1().Pods(k.namespace.Name)
+	pods, err := k8sPods.List(context.Background(), metaV1.ListOptions{
+		LabelSelector: k.releaseSelector(),
+	})
+	if err != nil {
+		return err
+	}
+	for _, p := range pods.Items {
+		ports, err := forwardPodPorts(&p, k.k8sConfig, k.namespace.Name, k.stopChannels)
+		if err != nil {
+			return fmt.Errorf("unable to forward ports: %v", err)
+		}
+		k.ports = append(k.ports, ports...)
+		log.Info().Str("Manifest ID", k.id).Interface("Ports", ports).Msg("Forwarded ports")
+	}
+	return nil
+}
+
+func (k *K8sHelmManifest) WaitUntilHealthy() error {
+	// using helm wait, not needed
+	if err := k.forwardAllPodsPorts(); err != nil {
+		return err
+	}
+	if k.SetValuesHelmFunc != nil {
+		if err := k.SetValuesHelmFunc(k); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (k *K8sHelmManifest) releaseSelector() string {
+	return fmt.Sprintf("%s=%s", ReleasePrefix, k.releaseName)
+}
+
+func (k *K8sHelmManifest) getServices() ([]coreV1.Service, error) {
+	svcs := k.k8sClient.CoreV1().Services(k.namespace.Name)
+	svcList, err := svcs.List(context.Background(), metaV1.ListOptions{
+		LabelSelector: k.releaseSelector(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	return svcList.Items, nil
+}
+
+func (k *K8sHelmManifest) ServiceDetails() ([]*ServiceDetails, error) {
+	sd := make([]*ServiceDetails, 0)
```

Comment:
This function can just be the same as the original manfiest deployer? All it's doing is looping over the ports forwarded.